sw/host/opentitantool/src/command/spi.rs

```diff
@@ -0,0 +1,263 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use erased_serde::Serialize;
+use progressing::mapping::Bar as MappingBar;
+use progressing::Baring;
+use std::fs::{self, File};
+use std::io::{self, Write};
+use std::time::Instant;
+use structopt::StructOpt;
+
+use opentitanlib::app::command::CommandDispatch;
+use opentitanlib::io::spi::Transfer;
+use opentitanlib::spiflash::SpiFlash;
+use opentitanlib::transport::{Capability, Transport};
+
+const PROGRESS_BAR_LEN: usize = 50;
+
+/// Read and parse an SFDP table.
+#[derive(Debug, StructOpt)]
+pub struct SpiSfdp {
+    #[structopt(
+        short,
+        long,
+        help = "Display raw SFDP bytes rather than the parsed struct."
+    )]
+    raw: Option<usize>,
+}
+
+fn hexdump(buf: &[u8]) {
```

Comment:
What does this function do? It's a little hard to tell so I"m not sure how to advise writing it in a simpler fashion.

Reply:
It prints a hexdump of a buffer in "exemplary form".  I added some comments explaining what's going on.